### PR TITLE
INTDEV-432 policy checks and denied operations

### PR DESCRIPTION
**Pull request by samutoljamo-pareto**

### CODE REVIEW

File: content/defaultReport/.schema
Comment (Fuzzbender) [inline content/defaultReport/.schema:1]: This was already fixed in INTDEV-513.

File: tools/app/app/api/card/[key]/route.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    });
+  }
+  try {
+    const calculate = new Calculate();
+    await calculate.generate(projectPath);
+    const card = await calculate.runQuery(projectPath, 'card', {
+      cardKey: params.key,
+    });
+    if (card.error) {
+      throw new Error(card.error);
```
Comment (Fuzzbender) [inline tools/app/app/api/card/[key]/route.tsx:53]: Why not throw directly from `calculate.runQuery`?

File: tools/app/app/cards/[key]/page.tsx [hunk 2]
```diff
@@ -42,6 +51,10 @@ export default function Page({ params }: { params: { key: string } }) {
 
   const [linksVisible, setLinksVisible] = useState(false);
 
+  useEffect(() => {
+    console.log(cardQuery);
```
Comment (Fuzzbender) [inline tools/app/app/cards/[key]/page.tsx:55]: Is this - for now - for debugging?

File: tools/app/app/cards/[key]/page.tsx [hunk 3]
```diff
@@ -54,80 +67,86 @@ export default function Page({ params }: { params: { key: string } }) {
...
 
-  if (error) {
+  if (error || errorCardQuery) {
     let errorMessage = t('unknownError');
     if (error instanceof Error) {
-      errorMessage = (error as Error).message;
+      errorMessage = error.message;
+    }
+    if (errorCardQuery instanceof Error) {
+      errorMessage = errorCardQuery.message;
```
Comment (Fuzzbender) [inline tools/app/app/cards/[key]/page.tsx:76]: This now overwrites if `errorMessage` was already populated by `error.message`. Is query error always more meaningful?

File: tools/app/app/components/ContentArea.tsx [hunk 4]
```diff
@@ -29,6 +29,9 @@ import {
   Option,
   Autocomplete,
   ChipDelete,
+  List,
+  ListItem,
+  Alert,
```
Comment (Fuzzbender) [inline tools/app/app/components/ContentArea.tsx:34]: It would be helpful if the imports from single source would be in alphabetical order.

**Review by Fuzzbender — APPROVED** [review]

File: content/defaultReport/.schema
Comment (samutoljamo-pareto) [reply content/defaultReport/.schema:1]: Rebased

File: tools/app/app/cards/[key]/page.tsx (hunk 2 above)
Comment (samutoljamo-pareto) [reply tools/app/app/cards/[key]/page.tsx:55]: Removed

File: tools/app/app/components/ContentArea.tsx (hunk 4 above)
Comment (samutoljamo-pareto) [reply tools/app/app/components/ContentArea.tsx:34]: Done

File: tools/app/app/cards/[key]/page.tsx (hunk 3 above)
Comment (samutoljamo-pareto) [reply tools/app/app/cards/[key]/page.tsx:72]: I'll leave it like this for now. Soon, cardQuery will replace the other endpoint

File: tools/app/app/api/card/[key]/route.tsx (hunk 1 above)
Comment (samutoljamo-pareto) [reply tools/app/app/api/card/[key]/route.tsx]: Done

**Pull request merged**